src/features/courses/course-controller.ts

```diff
@@ -3717,18 +3703,46 @@ You should be able to reply to the student's email address (${options.student.pr
         // eslint-disable-next-line @typescript-eslint/no-explicit-any
         const data: any = mainData?.get({plain: true});
 
-        await mainData?.questions?.asyncForEach(async (question, i) => 
+        await mainData?.questions?.asyncForEach(async (question, i) =>
             await question.grades?.asyncForEach(async (grade, j) => {
-                if (_.isNil(grade.lastInfluencingAttempt) || _.isNil(grade.lastInfluencingAttempt.studentGradeInstance)) {
-                    logger.debug('No lastInfluencingAttempt and no studentGradeInstance exists');
+                if (_.isNil(grade.lastInfluencingCreditedAttemptId)) {
+                    console.log(data.questions[i].grades[j]);
+                    logger.error('No lastInfluencingCreditedAttemptId. Cannot find the best version.');
                     return;
                 }
-                const probAttach = await grade.lastInfluencingAttempt.studentGradeInstance.getProblemAttachments();
-                grade.lastInfluencingAttempt.studentGradeInstance.problemAttachments = probAttach;
-                data.questions[i].grades[j].lastInfluencingAttempt.studentGradeInstance.problemAttachments = probAttach.map(pA => pA.get({plain: true}));
+
+                const gradeInstanceAttachments = await StudentWorkbook.findOne({
```

Comment:
Why not us a mixin 🤷‍♂️

Reply:
I gave the mixin a try, and it didn't pull anything back. I updated the PR with the mixin function.